src/libraries/System.Memory.Data/src/System/IO/ReadOnlyMemoryStream.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
We already have one of these here:
https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/IO/ReadOnlyMemoryStream.cs
Let's reuse the same one and avoid having two.

Reply:
This copy has ValidateNotClosed calls that the existing one doesn't; so unifying them would require one of

* Removing the "I'm disposed already" exceptions from here
* Adding the "I'm disposed already" exceptions to the existing one (which, IIRC, is never returned to user code and why it skipped them)
* Something with \#if.

Of those, I'd go for adding the ObjectDisposedException to the current copy; but since it's currently only used for some internal networking state (IIRC) there may be conflicting opinions.

Reply:
The existing one can actually find its way out to user code, but I'm fine with us adding exceptions on disposed.

Reply:
I had to do some work on the common source in order to add the ValidateNotClosed and to make it cross-compile for netstandard2.0.  Stay tuned for the update.